bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirHandler.java

```diff
@@ -0,0 +1,290 @@
+        ComfoAirSerialConnector comfoAirConnector = this.comfoAirConnector;
+
+        if (comfoAirConnector != null) {
+            Integer requestCmd = command.getRequestCmd();
+            Integer replyCmd = command.getReplyCmd();
+            int[] requestData = command.getRequestData();
+
+            Integer preRequestCmd;
+            Integer preReplyCmd;
+            int[] preResponse = new int[0];
+
+            if (requestCmd != null) {
+                switch (requestCmd) {
+                    case 0x9f:
+                        preRequestCmd = 0x9d;
+                        preReplyCmd = 0x9e;
+                        break;
+                    case 0xcb:
+                        preRequestCmd = 0xc9;
+                        preReplyCmd = 0xca;
+                        break;
+                    case 0xcf:
+                        preRequestCmd = 0xcd;
+                        preReplyCmd = 0xce;
+                        break;
+                    case 0xd7:
+                        preRequestCmd = 0xd5;
+                        preReplyCmd = 0xd6;
+                        break;
+                    case 0xed:
+                        preRequestCmd = 0xeb;
+                        preReplyCmd = 0xec;
+                        break;
+                    default:
+                        preRequestCmd = requestCmd;
+                        preReplyCmd = replyCmd;
+                }
+
+                if (!preRequestCmd.equals(requestCmd)) {
+                    command.setRequestCmd(preRequestCmd);
+                    command.setReplyCmd(preReplyCmd);
+                    command.setRequestData(new int[0]);
+
+                    preResponse = comfoAirConnector.sendCommand(command, new int[0]);
+
+                    if (preResponse.length <= 0) {
+                        return UnDefType.NULL;
+                    } else {
+                        command.setRequestCmd(requestCmd);
+                        command.setReplyCmd(replyCmd);
+                        command.setRequestData(requestData);
+                    }
+                }
+
+                int[] response = comfoAirConnector.sendCommand(command, preResponse);
+
+                if (response.length > 0) {
+                    ComfoAirCommandType comfoAirCommandType = ComfoAirCommandType.getCommandTypeByKey(commandKey);
+
+                    ComfoAirDataType dataType = comfoAirCommandType.getDataType();
+                    State value = dataType.convertToState(response, comfoAirCommandType);
+
+                    if (value == null) {
+                        logger.warn("unexpected value for DATA: {}", ComfoAirSerialConnector.dumpData(response));
+                        return UnDefType.UNDEF;
+                    } else {
+                        return value;
+                    }
+                }
+            }
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private class AffectedItemsUpdateThread extends Thread {
```

Comment:
This class isn't used as a Thread at all, instead it is used as a Runnable.
```suggestion
    private class AffectedItemsUpdateThread implements Runnable {
```